### PR TITLE
Add the gnutls update to 3.7.2+ for SLE15-SP4

### PR DESCRIPTION
**Pull request by jouyingbin**
**Description:**
gnutls attempt to update the major version to 3.7.2+ in SLE15 SP4 base on the Jira feature

_Jira#SLE-[19765](https://jira.suse.com/browse/SLE-19765): Update libnettle and gnutls to new major versions_

This PR is also adding the version check and record_info/soft_failure for gnutls

- Related ticket: https://progress.opensuse.org/issues/102770
- Needles: NA
- Verification run: 
  https://openqa.suse.de/tests/7725188#step/gnutls_base_check/8 (x86_64 - KER mode)
  https://openqa.suse.de/tests/7725206#step/gnutls_base_check/8 (aarch64 - ENV mode)
  https://openqa.suse.de/tests/7725207#step/gnutls_base_check/8 (s390x - KER mode)

### CODE REVIEW

**Review by lilyeyes — APPROVED** [review]
LGTM

File: tests/fips/gnutls/gnutls_base_check.pm [hunk 1]
```diff
@@ -1,24 +1,36 @@
...
+# Maintainer: rfan1 <richard.fan@suse.com>, Ben Chou <bchou@suse.com>
+# Tags: poo#63223, poo#102770, tc#1744099
 
 use base 'consoletest';
 use testapi;
 use strict;
 use warnings;
 use utils 'zypper_call';
-use version_utils qw(is_tumbleweed);
+use version_utils qw(is_tumbleweed is_sle);
```
Comment (rfan1) [inline tests/fips/gnutls/gnutls_base_check.pm:16]: is_tumbleweed is not used, right? then we can remove

Comment (jouyingbin) [reply tests/fips/gnutls/gnutls_base_check.pm:16]: Good question, I think TW also apply this case. But I think my adjustment will not affect the TW test. So I think we should keep it there.

**Review by rfan1 — APPROVED** [review]
LGTM

**Comment by mloviska** [comment]
Hi Ben, it apparently breaks older sle versions [sle15sp3](https://openqa.suse.de/tests/7733480#step/gnutls_base_check/6)

**Pull request merged**